features/dd-sdk-android-session-replay-compose/src/test/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/SliderSemanticsNodeMapperNodeMapperTest.kt

```diff
@@ -0,0 +1,155 @@
+
+    @Mock
+    private lateinit var mockSemanticsNode: SemanticsNode
+
+    @Mock
+    private lateinit var mockAsyncJobStatusCallback: AsyncJobStatusCallback
+
+    @Mock
+    private lateinit var mockProgressBarRangeInfo: ProgressBarRangeInfo
+
+    @Mock
+    private lateinit var mockRange: ClosedFloatingPointRange<Float>
+
+    @LongForgery(min = 0xffffffff)
+    var fakeBackgroundColor: Long = 0L
+
+    @FloatForgery
+    var fakeCornerRadius: Float = 0f
+
+    @FloatForgery
+    var fakeStart: Float = 0f
+
+    @FloatForgery
+    var fakeCurrent: Float = 0f
+
+    @FloatForgery
+    var fakeEndInclusive: Float = 0f
+
+    @Forgery
+    lateinit var fakeUiContext: UiContext
+
+    @BeforeEach
+    override fun `set up`(forge: Forge) {
+        super.`set up`(forge)
+        testedSliderSemanticsNodeMapper = SliderSemanticsNodeMapper(
+            colorStringFormatter = mockColorStringFormatter,
+            semanticsUtils = mockSemanticsUtils
+        )
+    }
+
+    private fun mockSemanticsNode(): SemanticsNode {
+        return mockSemanticsNodeWithBound {
+            whenever(mockSemanticsNode.layoutInfo).doReturn(mockLayoutInfo)
+        }
+    }
+
+    fun `M return the correct wireframe W map`() {
+        // Given
+        val mockSemanticsNode = mockSemanticsNode()
+        val fakeGlobalBounds = rectToBounds(fakeBounds, fakeUiContext.density)
+        val fakeBackgroundInfo = BackgroundInfo(
+            globalBounds = fakeGlobalBounds,
+            color = fakeBackgroundColor,
+            cornerRadius = fakeCornerRadius
+        )
+        whenever(mockSemanticsUtils.resolveInnerBounds(mockSemanticsNode)) doReturn fakeGlobalBounds
+        whenever(mockSemanticsUtils.resolveBackgroundInfo(mockSemanticsNode)) doReturn listOf(
+            fakeBackgroundInfo
+        )
+        whenever(mockSemanticsUtils.getProgressBarRangeInfo(mockSemanticsNode)) doReturn
+            mockProgressBarRangeInfo
+        whenever(mockProgressBarRangeInfo.range) doReturn mockRange
+        whenever(mockProgressBarRangeInfo.current) doReturn fakeCurrent
+        whenever(mockRange.start) doReturn fakeStart
+        whenever(mockRange.endInclusive) doReturn fakeEndInclusive
+
+        // When
+        val actual = testedSliderSemanticsNodeMapper.map(
+            mockSemanticsNode,
+            fakeUiContext,
+            mockAsyncJobStatusCallback
+        )
+
+        // Then
+        val trackHeight = DEFAULT_TRACK_HEIGHT.value * fakeUiContext.density
+        val yTrackOffset = (fakeGlobalBounds.height - trackHeight).toLong() / 2
+        val fakeProgress = fakeCurrent / (fakeEndInclusive - fakeStart)
+        val xOffset = fakeProgress * fakeGlobalBounds.width + fakeGlobalBounds.x
+        val thumbHeight = DEFAULT_THUMB_RADIUS.value * 2 * fakeUiContext.density
+        val expectedTrackWireframe = MobileSegment.Wireframe.ShapeWireframe(
+            id = (fakeSemanticsId.toLong() shl 32) + 0,
```

Comment:
> [!NOTE]
> It's not clear why we need shift here?

Reply:
Compose requires `Ulong` color value, so we need to shift 32 bit here, it explains in prod code.